src/pages/instances/forms/CreateInstanceFromSnapshotForm.tsx

```diff
@@ -0,0 +1,308 @@
+        (instance.devices["root"] as LxdDiskDevice)?.pool ??
+        storagePools[0]?.name,
+    },
+    enableReinitialize: true,
+    validate,
+    validationSchema: Yup.object().shape({
+      instanceName: instanceNameValidation(
+        instance.project,
+        controllerState,
+        instance.name,
+      ).required(),
+    }),
+    onSubmit: (values) => {
+      createInstance(
+        JSON.stringify(instanceFromSnapshotPayload(values, instance, snapshot)),
+        values.targetProject,
+        values.targetClusterMember,
+      )
+        .then((operation) => {
+          toastNotify.info(
+            `Instance creation started for ${values.instanceName}.`,
+          );
+          eventQueue.set(
+            operation.metadata.id,
+            () => notifySuccess(values.instanceName, values.targetProject),
+            (msg) =>
+              toastNotify.failure(
+                "Instance creation failed.",
+                new Error(msg),
+                <InstanceLink instance={instance} />,
+              ),
+          );
+        })
+        .catch((e) => {
+          toastNotify.failure(
+            "Instance creation failed.",
+            e,
+            <InstanceLink instance={instance} />,
+          );
+        })
+        .finally(() => {
+          close();
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={close}
+      className="duplicate-instances-modal"
+      title="Create instance from snapshot"
+      buttonRow={
+        <>
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={close}
+          >
+            Cancel
+          </Button>
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            loading={formik.isSubmitting}
+            disabled={
+              !formik.isValid ||
+              storagePoolsLoading ||
+              clusterMembersLoading ||
+              projectsLoading
+            }
+            onClick={() => void formik.submitForm()}
+          >
+            Create
+          </ActionButton>
+        </>
+      }
+    >
+      <Form onSubmit={formik.handleSubmit}>
+        <Input
+          type="text"
+          label="Snapshot name"
+          disabled
+          value={snapshot.name}
+        />
+        <Input
+          {...formik.getFieldProps("instanceName")}
+          type="text"
+          label="New instance name"
+          error={formik.errors.instanceName}
+        />
+        {isClustered && (
+          <Select
+            {...formik.getFieldProps("targetClusterMember")}
+            id="targetClusterMember"
+            label="Target cluster member"
+            options={clusterMembers.map((clusterMember) => {
+              return {
+                label: clusterMember.server_name,
+                value: clusterMember.server_name,
+              };
+            })}
+          />
```

Comment:
Should we create a reusable `ClusterMemberSelector`? We could also use it in the `DuplicateInstanceForm`.

Reply:
done and implemented in `DuplicateInstanceForm`